src/serializer/ResidualHeapSerializer.js

```diff
@@ -1101,6 +1101,13 @@ export class ResidualHeapSerializer {
                   mightHaveBeenDeleted
                 );
               });
+
+              // Although the property needs to be delayed, we still want to emit "undefined"
+              // as part of the object literal to ensure a consistent property ordering.
+              if (!mightHaveBeenDeleted) {
```

Comment:
I think this test case brings out the issue:
```js
function() {
    let x = global.__abstract ? __abstract("boolean", "false") : false;
    let o = { a: undefined, b: {} };
    o.a = o;
    if (x) delete o.a;

    global.inspect = function() {
      for (let name in o) { return name; }
    };
})();
```

Can you either address it as described above, or create a new issue to track the remaining work?